src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -780,9 +821,154 @@ protected static void clipForOverlappingReads(final SAMRecord read1, final SAMRe
                         final int clipFrom = neg.getReadLength() - negDiff - clipped + 1;
                         CigarUtil.softClip3PrimeEndOfRead(neg, Math.min(neg.getReadLength(), clipFrom));
                     }
+
+                }
+            }
+        }
+    }
+
+    protected static int getReadPositionToClipFrom(final SAMRecord rec, final int refPosToClipFrom) {
+
+        if (refPosToClipFrom > rec.getUnclippedEnd() || refPosToClipFrom < rec.getUnclippedStart()) {
+            //read doesn't cover position
+            return -1;
+        }
+        final Cigar cigar = rec.getCigar();
+        List<CigarElement> cigarElements = new ArrayList<>(cigar.getCigarElements()); //need to be modifiable
+        if (rec.getReadNegativeStrandFlag()) {
+            Collections.reverse(cigarElements);
+        }
+
+        int currentRefPosition = rec.getReadNegativeStrandFlag()? -rec.getUnclippedEnd() : rec.getUnclippedStart();
+        final int refPosObjective = rec.getReadNegativeStrandFlag()? -refPosToClipFrom : refPosToClipFrom;
+
+        int currentReadPosition = 1;
+
+        for (final CigarElement cigarElement : cigarElements) {
+
+            final CigarOperator op = cigarElement.getOperator();
+
+            if ((op.consumesReferenceBases() || op.isClipping()) && currentRefPosition + cigarElement.getLength() >= refPosObjective) {
+                if (op.consumesReadBases() || op == CigarOperator.SOFT_CLIP) {
+                    return currentReadPosition + refPosObjective - currentRefPosition;
+                } else if (op == CigarOperator.HARD_CLIP) {
+                    //for hardclip, position is not on read
+                    return -1;
+                } else {
+                    //if doesn't consume read bases and isn't clipping, return current base
+                    return currentReadPosition;
                 }
             }
+
+            if (op.consumesReferenceBases() || op.isClipping()) {
+                currentRefPosition += cigarElement.getLength();
+            }
+
+            if (op.consumesReadBases() || op.isClipping()) {
+                currentReadPosition += cigarElement.getLength();
+            }
         }
+
+        //if we haven't found the correct position, there is a bug
+        throw new PicardException("Could not find read position for reference position " + refPosToClipFrom + " which should be covered by read " + rec);
+    }
+
+    protected static void clipForOverlappingReads(final SAMRecord read1, final SAMRecord read2) {
+        clipForOverlappingReads(read1, read2, CigarOperator.SOFT_CLIP, null, null, Collections.emptyList());
+    }
+
+    protected static boolean basesToClipMatchAdapterPair(final SAMRecord read, final int posToClip, final AdapterPair adapterPair,
```

Comment:
This function is awfully complex....but it seems to be doing a combination of things for which other functions already exist...can you explain what it does? perhaps it needs to be broken up into a few smaller methods.